drivers/flash/flash_rpc/flash_rpc_host.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/logging/log.h>
+#include <drivers/flash/flash_rpc.h>
+
+#include <nrf_rpc_cbor.h>
+#include <nrf_rpc/nrf_rpc_ipc.h>
+
+#include <zcbor_common.h>
+#include <zcbor_decode.h>
+#include <zcbor_encode.h>
+
+
+LOG_MODULE_DECLARE(FLASH_RPC, CONFIG_FLASH_RPC_LOG_LEVEL);
+
+#define CBOR_BUF_FLASH_MSG_SIZE (sizeof(void *) + sizeof(size_t) + sizeof(off_t))
+
+NRF_RPC_IPC_TRANSPORT(flash_rpc_api_tr, DEVICE_DT_GET(DT_NODELABEL(ipc0)), "flash_rpc_api_ept");
+NRF_RPC_GROUP_DEFINE(flash_rpc_api, "flash_rpc_api", &flash_rpc_api_tr, NULL, NULL, NULL);
+
+static const struct device *const flash_controller =
+	DEVICE_DT_GET_OR_NULL(DT_CHOSEN(zephyr_flash_controller));
+
+static bool decode_flash_msg(struct nrf_rpc_cbor_ctx *ctx, off_t *offset, void **ptr, size_t *len)
+{
+	uint32_t ptr_addr;
+
+	if (!zcbor_uint32_decode(ctx->zs, (uint32_t *)offset)) {
+		return false;
+	}
+	if (zcbor_uint32_decode(ctx->zs, &ptr_addr)) {
+		*ptr = (void *)ptr_addr;
+	} else {
+		return false;
+	}
+
+	return zcbor_uint32_decode(ctx->zs, (uint32_t *)len);
+}
+
+static void rsp_error_code_send(const struct nrf_rpc_group *group, int err_code)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(group, ctx, sizeof(int));
+
+	zcbor_int32_put(ctx.zs, err_code);
+
+	nrf_rpc_cbor_rsp_no_err(group, &ctx);
+}
+
+static void flash_rpc_init_handler(const struct nrf_rpc_group *group,
```

Comment:
No other function checks whether this has been called so I guess this is optional.

Reply:
OK, now I see that this is called on init by the controller side.